### DISCUSSION TITLE
Git LFS + Github pages: any plans to support?

### DISCUSSION THREAD

**Discussion by dy** [community/community#50337] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

It seems Github Pages doesn't work with git LFS. Many developers post issues to wrong repositories: [1](https://github.com/git-lfs/git-lfs/issues/2725), [2](https://github.com/git-lfs/git-lfs/issues/1342), [3](https://github.com/git-lfs/git-lfs/issues/3498), [4](https://stackoverflow.com/questions/56470398/mp4-file-stored-with-git-lfs-not-playing-in-github-pages).

Personally I have a use case to have a collection of assets, that must be accessible both from an app and from website, eg [this](https://media.githubusercontent.com/media/krishnized/japp/main/audio/Srila-Prabhupada-1.mp3).

The problem is that browser throws 

```
Srila-Prabhupada-1.mp3:1 Refused to load media from 'https://media.githubusercontent.com/media/krishnized/japp/main/audio/Srila-Prabhupada-1.mp3' because it violates the following Content Security Policy directive: "default-src 'none'". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.
```

Is there any consideration from pages team to enable proper mime type detection for LFS?

It would be unfortunate to force users to limit media files to 25Mb and store hundreds of them in repository just to workaround that github pages limitation. Or worse - to host files from somewhere else.

**Comment by yoannchaudet** [comment]
There is no plans to support Git LFS in GitHub Pages. We actually don't serve Pages' static assets from Git repositories.

Now that [we support deploying Pages sites from Actions](https://github.blog/changelog/2022-07-27-github-pages-custom-github-actions-workflows-beta/), "enabling LFS" should be possible with:

1. Change your site's source from Branch to GitHub Actions in the settings
2. Pick a workflow from our starter workflows (we suggest them but they are also [here](https://github.com/actions/starter-workflows/tree/main/pages))
3. Finally, update your workflow to update the `actions/checkout` step ([example](https://github.com/actions/starter-workflows/blob/20496ebfca42582d0ca772172a927616bc3bcd41/pages/static.yml#L32-L33)) to enable LFS

   ```yaml
         - name: Checkout
           uses: actions/checkout@v3
           with:
             lfs: true
   ```

**Reply by dy** [reply]
Ok, not sure I understood the actions proposal. Does that mean I can store media as git lfs but somehow serve it from github pages?

**Reply by yoannchaudet** [reply]
correct

**Reply by dy** [reply]
Ok, seemingly every deploy of LFS such way uploads all audios, which is very slow, but also uses LFS storage. Yesterday after 3 commits github exceeded LFS storage for the organization. I suppose deploying this way is not the most efficient to host media for github pages.

Now I have to delete the repository.
Will be looking for better method or store smaller chunks straight in repository.

**Reply by marcindulak** [reply]
For public repos, in the case when using Jekyll is OK, there is a workaround to use a custom Jekyll filter to convert the relative GitHub LFS resource paths to full GitHub URLs during Jekyll build. This saves on LFS operations, since it explicitly uses `lfs: false` during the checkout (so the clone behaves effectively as `GIT_LFS_SKIP_SMUDGE=1 git clone`). See a demo at https://github.com/marcindulak/jekyll-github-lfs
